### PR TITLE
GV-2 Add GoRouter and check scannedText is null or not

### PR DESCRIPTION
**Pull request by f16f16f16f16**
- Add GoRouter
- Check scannedText is null or not

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -207,17 +225,19 @@ class QRCodeScannerScreen extends StatelessWidget {
               final Uint8List? image = capture.image;
               for (final barcode in barcodes) {
                 final scannedText = barcode.rawValue;
-                if (scannedText!
-                    .startsWith('https://viewer.graffity.app/ar/')) {
+                if (scannedText == null) {
+                  continue; // Skip this if scannedText is null
+                }
+                if (scannedText.startsWith('https://viewer.graffity.app/ar/')) {
```
Comment (f16f16f16f16) [inline lib/main.dart:231]: check scannedText is null or not

File: lib/main.dart [hunk 2]
```diff
@@ -4,13 +4,29 @@ import 'package:flutter/material.dart';
...
+      routes: [
+        GoRoute(
+          path: 'details',
+          builder: (_, __) => Scaffold(
+            appBar: AppBar(title: const Text('Details Screen')),
+          ),
+        ),
+      ],
+    ),
+  ],
```
Comment (f16f16f16f16) [inline lib/main.dart:28]: Create a `GoRouter` to handle the routing

File: lib/main.dart [hunk 3]
```diff
@@ -3,10 +3,30 @@ import 'dart:io';
...
+);
```
Comment (f16f16f16f16) [inline lib/main.dart:29]: Create a GoRouter to handle the routing

**Review by bankbiz — APPROVED** [review]
LGTM

**Pull request merged**